scipy/signal/bspline_util.c.in

```diff
@@ -4,66 +4,90 @@
 #include <stdlib.h>
 #include <string.h>
 #include <stdio.h>
+
 #define NO_IMPORT_ARRAY
 #include "numpy/arrayobject.h"
 
-void compute_root_from_lambda(double, double *, double *);
-
 #ifndef M_PI
 #define M_PI           3.14159265358979323846  /* pi */
 #endif
 
-#define CONJ(a) ((a))
-#define ABSQ(a) ( (a*CONJ(a)))
+#define CONJ(a) (~((__complex__)a))
+#define ABSQ(a) (__real__ (a*CONJ(a)))
+
+void
+compute_root_from_lambda(lambda, r, omega)
+     double lambda;
+     double *r;
+     double *omega;
+{
+    double xi;
+    double tmp, tmp2;
+
+    tmp = sqrt(3 + 144*lambda);
+    xi = 1 - 96*lambda + 24*lambda * tmp;
+    *omega = atan(sqrt((144*lambda - 1.0)/xi));
+    tmp2 = sqrt(xi);
+    *r = (24*lambda - 1 - tmp2)/(24*lambda)	* sqrt((48*lambda + 24*lambda*tmp))/tmp2;
+    return;
+}
+
+
+{{py:
+
+CNAME = ['C', 'Z']
+CTYPE = ['__complex__ float', '__complex__ double']
+
+NAME = ['D', 'S']
```

Comment:
Please declare these as `RNAME` and `RTYPE` and form `NAMES`, `TYPES` vars as the sum of `R` and `C` variants. Then you can use them in the code below instead of doing the `CNAME+NAME` preparations at the top of each loop.